src/DI/DoctrineExtensionsExtension.php

```diff
@@ -136,6 +136,7 @@ public function loadConfiguration(): void
 			$builder->addDefinition($this->prefix('ipTraceable'))
 				->setFactory(IpTraceableListener::class)
 				->addSetup('setAnnotationReader', ['@' . Reader::class])
+				->addSetup('setIpValue', ['@Nette\Http\Request::remoteAddress'])
```

Comment:
nettrine/extensions does not require nette/http, this is hidden dependency.

Reply:
Hmm, it's true. What do you think, if I add new setting key, e.g. IpCallback

Reply:
Imho too specific, you can use service configuration

```
services:
  nettrine.extensions.ipTraceable:
    setup:
      - setIpValue('@Nette\Http\Request::remoteAddress')
```

Reply:
OK, I thought it would be good. This setting/method (setIpValue) is "invisible" (not documented) and extension does not work without it.

Reply:
I see. Let's wait for @f3l1x opinion, in case it's required may be configuration option acceptable.

Reply:
Another option is add new information row and your solution to documentation :)

Reply:
Well, I wouldn't add a new dependency. What about to use `$_SERVER['REMOTE_ADDR']`? 

It's basically what nette/http pkg does (https://github.com/nette/http/blob/master/src/Http/RequestFactory.php#L259). It wont work with load balancer, but in that case we can put this hint into doc.

Hm?

Reply:
Yes, I think that's enough ;)

Reply:
We may also enforce configuration of setIpValue by schema, if ipTraceable is enabled

Reply:
Something like that
`Expect::anyof(false, Expect::structure(['ipValue' => Expect::string()]))->default(false);`